applications/StructuralMechanicsApplication/custom_constitutive/linear_elastic_orthotropic_2D_law.hpp

```diff
@@ -0,0 +1,281 @@
+		* @param rFeatures
+		*/
+		void GetLawFeatures(Features& rFeatures); //update this
+
+		bool Has(const Variable<double>& rThisVariable);
+		bool Has(const Variable<Vector>& rThisVariable);
+		bool Has(const Variable<Matrix>& rThisVariable);
+
+		double& GetValue(const Variable<double>& rThisVariable, double& rValue);
+		Vector& GetValue(const Variable<Vector>& rThisVariable, Vector& rValue);
+		Matrix& GetValue(const Variable<Matrix>& rThisVariable, Matrix& rValue);
+
+		void SetValue(const Variable<double>& rVariable,
+			const double& rValue,
+			const ProcessInfo& rCurrentProcessInfo);
+		void SetValue(const Variable<Vector>& rThisVariable,
+			const Vector& rValue,
+			const ProcessInfo& rCurrentProcessInfo);
+		void SetValue(const Variable<Matrix>& rThisVariable,
+			const Matrix& rValue,
+			const ProcessInfo& rCurrentProcessInfo);
+		/**
+		* Material parameters are inizialized
+		*/
+		void InitializeMaterial(const Properties& rMaterialProperties,
+			const GeometryType& rElementGeometry,
+			const Vector& rShapeFunctionsValues);
+
+		void InitializeSolutionStep(const Properties& rMaterialProperties,
+			const GeometryType& rElementGeometry, //this is just to give the array of nodes
+			const Vector& rShapeFunctionsValues,
+			const ProcessInfo& rCurrentProcessInfo);
+
+		void FinalizeSolutionStep(const Properties& rMaterialProperties,
+			const GeometryType& rElementGeometry, //this is just to give the array of nodes
+			const Vector& rShapeFunctionsValues,
+			const ProcessInfo& rCurrentProcessInfo);
+
+		/**
+		* Computes the material response:
+		* PK2 stresses and algorithmic ConstitutiveMatrix
+		* @param rValues
+		* @see   Parameters
+		*/
+		void CalculateMaterialResponsePK2(Parameters & rValues);
+
+		/**
+		* Computes the material response:
+		* Kirchhoff stresses and algorithmic ConstitutiveMatrix
+		* @param rValues
+		* @see   Parameters
+		*/
+		//void CalculateMaterialResponseKirchhoff(Parameters & rValues);
+
+		/**
+		* This function is designed to be called once to perform all the checks needed
+		* on the input provided. Checks can be "expensive" as the function is designed
+		* to catch user's errors.
+		* @param rMaterialProperties
+		* @param rElementGeometry
+		* @param rCurrentProcessInfo
+		* @return
+		*/
+		int Check(const Properties& rMaterialProperties, const GeometryType& rElementGeometry, const ProcessInfo& rCurrentProcessInfo);
+
+		/**
+		* Input and output
+		*/
+		/**
+		* Turn back information as a string.
+		*/
+		//virtual String Info() const;
+		/**
+		* Print information about this object.
+		*/
+		//virtual void PrintInfo(std::ostream& rOStream) const;
+		/**
+		* Print object's data.
+		*/
+		//virtual void PrintData(std::ostream& rOStream) const;
+
+	protected:
+
+		///@name Protected static Member Variables
+		///@{
+		///@}
+		///@name Protected member Variables
+		///@{
+		Matrix mInverseDeformationGradientF0;
```

Comment:
...already commented...but please do not store anything here, unless you REALLY need to